frontend/lib/src/components/legend_section.dart

```diff
@@ -0,0 +1,67 @@
+import 'package:flutter/material.dart';
+import 'package:get/get.dart';
+
+import 'package:frontend/src/components/indicator.dart';
+import 'package:frontend/src/controllers/spending_breakdown_controller.dart';
+import 'package:frontend/src/enums/transaction_enum.dart';
+import 'package:frontend/src/models/spending_breakdown.dart';
+
+/*
+  A Column widget where transaction data displayed in graph is explained
+
+  @param spendingList: List of SpendingBreakdown to figure out which legends to display
+*/
+
+class LegendSection extends StatelessWidget {
+  const LegendSection({super.key, required this.spendingList});
+
+  final List<SpendingBreakdown> spendingList;
+
+  @override
+  Widget build(BuildContext context) {
+    final SpendingBreakdownController spendingBreakdownController = Get.find();
+    List<Widget> legends = [];
```

Comment:
pls decide which one technique to use, pass by parameters or get by controller
for `spendingList` and `totalSpent`